File: sample-conference-videochat/sample-conference-videochat/Components/ChatScreen/SelectAssetsScreen/SelectAssetCell.m
//
//  SelectAssetCell.m
//  sample-conference-videochat
//
//  Created by Injoit on 2/6/20.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "SelectAssetCell.h"
#import "UIView+Chat.h"

@implementation SelectAssetCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.checkBoxImageView.hidden = YES;
    self.checkBoxView.backgroundColor = [UIColor.whiteColor colorWithAlphaComponent:0.35f];
    [self.checkBoxView setRoundBorderEdgeColorView:4.0f borderWidth:1.0f color:nil borderColor:UIColor.whiteColor];
    [self.videoTypeView setRoundViewWithCornerRadius:3.0f];
    self.videoTypeView.hidden = YES;
    self.assetImageView.contentMode = UIViewContentModeScaleAspectFill;
}

- (void)prepareForReuse {
    [super prepareForReuse];
    
    self.assetTypeImageView.image = nil;
    self.checkBoxView.backgroundColor = [UIColor.whiteColor colorWithAlphaComponent:0.35f];
    [self.checkBoxView setRoundBorderEdgeColorView:4.0f borderWidth:1.0f color:nil borderColor:UIColor.whiteColor];
    self.videoTypeView.hidden = YES;
    self.checkBoxImageView.hidden = YES;
}

- (void)setSelected:(BOOL)selected {
    [self onSelectedCell:selected];
}

- (void)setHighlighted:(BOOL)highlighted {
    [self onSelectedCell:highlighted];
}

- (void)onSelectedCell:(Boolean)newValue {
    if (newValue == YES) {
        self.checkBoxImageView.hidden = NO;
        self.contentView.backgroundColor = [UIColor colorWithRed:0.85f green:0.89f blue:0.97f alpha:1.0f];
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
                                           borderWidth:1.0f
                                                 color:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]
                                           borderColor:[UIColor colorWithRed:0.22f green:0.47f blue:0.99f alpha:1.0f]];
    } else {
        self.checkBoxImageView.hidden = YES;
        self.contentView.backgroundColor = UIColor.clearColor;
        [self.checkBoxView setRoundBorderEdgeColorView:4.0f
        borderWidth:1.0f
              color:[UIColor.whiteColor colorWithAlphaComponent:0.35f]
        borderColor:UIColor.whiteColor];
    }
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/SelectAssetsScreen/SelectAssetCell.swift
//
//  SelectAssetCell.swift
//  sample-chat-swift
//
//  Created by Injoit on 12/9/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class SelectAssetCell: UICollectionViewCell {
    @IBOutlet weak var durationVideoLabel: UILabel!
    @IBOutlet weak var videoTypeView: UIView!
    @IBOutlet weak var assetTypeImageView: UIImageView!
    @IBOutlet weak var assetImageView: UIImageView!
    @IBOutlet weak var checkBoxImageView: UIImageView!
    @IBOutlet weak var checkBoxView: UIView!
    
    var representedAssetIdentifier: String!
    
    //MARK: - Overrides
    override func awakeFromNib() {
        checkBoxView.backgroundColor = UIColor.white.withAlphaComponent(0.35)
        checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                            borderWidth: 1.0,
                                            borderColor: UIColor.white)
        videoTypeView.rounded(cornerRadius: 3)
        videoTypeView.isHidden = true
        assetImageView.contentMode = .scaleAspectFill
    }
    
    override func prepareForReuse() {
        assetTypeImageView.image = nil
        checkBoxView.backgroundColor = UIColor.white.withAlphaComponent(0.35)
        checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                            borderWidth: 1.0,
                                            borderColor: UIColor.white)
        videoTypeView.isHidden = true
        checkBoxImageView.isHidden = true
    }
    
    override var isHighlighted: Bool {
        willSet {
            onSelectedCell(newValue)
        }
    }
    
    override var isSelected: Bool {
        willSet {
            onSelectedCell(newValue)
        }
    }
    
    func onSelectedCell(_ newValue: Bool) {
        if newValue == true {
            checkBoxImageView.isHidden = false
            contentView.backgroundColor = UIColor(red:0.85, green:0.89, blue:0.97, alpha:1)
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1),
                                                     borderColor: UIColor(red:0.22, green:0.47, blue:0.99, alpha:1))
        } else {
            checkBoxImageView.isHidden = true
            contentView.backgroundColor = .clear
            checkBoxView.setRoundBorderEdgeColorView(cornerRadius: 4.0,
                                                     borderWidth: 1.0,
                                                     color: UIColor.white.withAlphaComponent(0.35),
                                                     borderColor: UIColor.white)
        }
    }
}


